src/exchange/token_usage_collector.py

```diff
@@ -0,0 +1,40 @@
+from collections import defaultdict
+from dataclasses import dataclass
+import queue
+from typing import List
+
+from exchange.providers.base import Usage
+
+
+@dataclass
+class TokenUsage:
+    model: str
+    input_tokens: int
+    output_tokens: int
+
+
+class _TokenUsageCollector:
+    def __init__(self) -> None:
+        # use thread-safe queue to store usage data from multiple threads
+        # as data may be collected from multiple threads
+        self.usage_data_queue = queue.Queue()
+
+    def collect(self, model: str, usage: Usage) -> None:
+        self.usage_data_queue.put((model, usage.input_tokens, usage.output_tokens))
```

Comment:
i don't think we need a queue here (I don't see this making using of put/task_done etc in the sense of processing a queue in threads). If we're appending to a list (or adding to a running sum in a dictionary) the GIL will make sure all updates are processed, and I don't believe we have any concerns around ordering